app/controllers/api/physical_storages_controller.rb

```diff
@@ -20,6 +20,22 @@ def create_resource(_type, _id = nil, data = {})
       action_result(false, err.to_s)
     end
 
+    def delete_resource(type, id, _data = nil)
+      raise BadRequestError, "Must specify an id for deleting a #{type} resource" if id.blank?
+
+      ensure_resource_exists(type, id) if single_resource?
+
+      api_action(type, id) do |klass|
+        physical_storage = resource_search(id, type, klass)
```

Comment:
@llivne Re: my comment here https://github.com/ManageIQ/manageiq/pull/20713/files#r510124331

You can use `supports?(:delete)` here e.g.`raise physical_storage.unsupported_reason(:delete) unless physical_storage.supports?(:delete)`

Reply:
done